Main/ClientApp/src/Components/Dialogs/SpotifyLoginDialog.tsx

```diff
@@ -15,8 +15,23 @@ export default function SpotifyLoginDialog(props: SpotifyLoginDialogProps) {
   const title = "Login to Spotify";
   const description = "Login to spotify to add this song to queue";
   const authEndpoint = "https://accounts.spotify.com/authorize";
-  const clientId = "e9a69171037d403ba8f18aa96d36aa09"; //TODO: get from server
+  const [clientId, setClientId] = useState<string | null>(
+    SpotifyStore.getSpotifyClientId()
+  );
   const scopes = ["user-modify-playback-state"];
+  const spotifyRedirectUrl = `${authEndpoint}?client_id=${clientId}&redirect_uri=${
+    props.redirectUri
+  }&scope=${scopes.join("%20")}&response_type=token&show_dialog=true`;
```

Comment:
What's going on here?

Reply:
creating redirect link to spotify auth. taken from example

Reply:
If you can't tell what the code does, please include a link to where it is copied from